api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1509,6 +1535,107 @@ public void getObject(String bucketName, String objectName, String fileName)
+   *   byte[] buf = new byte[16384];
+   *   int bytesRead;
+   *   while ((bytesRead = stream.read(buf, 0, buf.length)) >= 0) {
+   *     System.out.println(new String(buf, 0, bytesRead));
+   *   }
+   *   stream.close();
+   * }
+   * </pre>
+   *
+   * @param bucketName
+   *          Bucket name.
+   * @param objectName
+   *          Object name in the bucket.
+   * @param encryptionMaterials
+   *          Encryption metadata details and operations.
+   *
+   * @return {@link InputStream} containing the object data.
+   *
+   * @throws InvalidBucketNameException
+   *           upon invalid bucket name is given
+   * @throws NoResponseException
+   *           upon no response from server
+   * @throws IOException
+   *           upon connection error
+   * @throws XmlPullParserException
+   *           upon parsing response xml
+   * @throws ErrorResponseException
+   *           upon unsuccessful execution
+   * @throws InternalException
+   *           upon internal library error
+   * @throws InvalidEncryptionMetadataException
+   *           upon encryption key/iv error
+   * @throws BadPaddingException
+   *           upon wrong padding
+   * @throws IllegalBlockSizeException
+   *           upon incorrect block size
+   * @throws NoSuchPaddingException
+   *           upon incorrect padding
+   * @throws InvalidAlgorithmParameterException
+   *           upon incorrect algorithm
+   */
+  public CipherInputStream getObject(String bucketName, String objectName, EncryptionMaterials encryptionMaterials)
+      throws InvalidBucketNameException, NoSuchAlgorithmException, InsufficientDataException, IOException,
+      InvalidKeyException, NoResponseException, XmlPullParserException, ErrorResponseException, InternalException,
+      InvalidArgumentException, InvalidEncryptionMetadataException, NoSuchPaddingException, IllegalBlockSizeException,
+      BadPaddingException, InvalidAlgorithmParameterException {
+
+    if (encryptionMaterials == null) {
+      throw new InvalidArgumentException("empty encryption properties not allowed");
+    }
+
+    // get encrypted object metadata and verify
+    ObjectStat stat = statObject(bucketName, objectName);
+
+    // check if encryption metadata is not present
+    if ((stat.contentKey() == null) || (stat.encryptionIv() == null)) {
+      throw new InvalidEncryptionMetadataException("encryption key or iv not present");
+    }
+
+    // Fetch encrypted object if metadata present
+    InputStream stream = getObject(bucketName, objectName);
+
+    // Get the encrypted key from response metadata
+    byte[] encryptedDataKey = Base64.decodeBase64(stat.contentKey().getBytes(StandardCharsets.UTF_8));
+
+    // Get the iv from the response metadata
+    byte[] iv = Base64.decodeBase64(stat.encryptionIv().getBytes(StandardCharsets.UTF_8));
+
+    // Decrypt the encrypted data key using master key
+    byte[] plainDataKey = encryptionMaterials.decryptContentKeys(encryptedDataKey);
```

Comment:
S3 spec calls it `data encryption key` or `data key` [(ref)](http://docs.aws.amazon.com/AmazonS3/latest/dev/UsingClientSideEncryption.html). I can rename the vars to `dataKey` and `encryptedDataKey`